File: backend/reference-manager--mit/src/main/java/com/umitakbulut/reference_manager/service/FlightService.java
package com.umitakbulut.reference_manager.service;

import com.umitakbulut.reference_manager.dto.request.FlightRequestDTO;
import com.umitakbulut.reference_manager.dto.response.FlightResponseDTO;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface FlightService {


    FlightResponseDTO addFlight(FlightRequestDTO requestDTO);

    FlightResponseDTO createFlight(FlightRequestDTO flightRequestDTO);

    FlightResponseDTO getFlightById(Long id);

    List<FlightResponseDTO> getAllFlights();

    FlightResponseDTO updateFlight(Long id, FlightRequestDTO flightRequestDTO);

    void deleteFlight(Long id);

    List<FlightResponseDTO> getArchivedFlights();

    void bulkUploadFlights(MultipartFile file);
}

File: frontend/src/services/flightService.js
import api from './api'

export const flightService = {
  getAllFlights: () => api.get('/api/v1/flights'),
  getFlightById: (id) => api.get(`/api/v1/flights/${id}`),
  createFlight: (flight) => api.post('/api/v1/flights', flight),
  updateFlight: (id, flight) => api.put(`/api/v1/flights/${id}`, flight),
  deleteFlight: (id) => api.delete(`/api/v1/flights/${id}`),
  addFlight: (flight) => api.post('/api/v1/flights', flight)
}
